control-plane/agents/src/bin/core/controller/registry.rs

```diff
@@ -100,6 +100,8 @@ pub(crate) struct RegistryInner<S: Store> {
     thin_args: ThinArgs,
     /// Check if the HA feature is enabled.
     ha_disabled: bool,
+    /// Max snapshots limit per volume.
+    max_snapshots: i32,
```

Comment:
Something like this or similar

```suggestion
    max_snapshots_volume: i32,
```